Library/ViewModels/PostCampaignCheckoutViewModel.swift

```diff
@@ -70,11 +83,21 @@ public class PostCampaignCheckoutViewModel: PostCampaignCheckoutViewModelType,
       context
     )
     .map { isLoggedIn, context in
-      // TODO: Calculate isEnabled and willRetryPaymentMethod fields instead of defaulting to true.
-      PledgeViewCTAContainerViewData(isLoggedIn, true, context, true)
+      PledgeViewCTAContainerViewData(
+        isLoggedIn: isLoggedIn,
+        isEnabled: true, // Pledge button never needs to be disabled on checkout page.
+        context: context,
+        willRetryPaymentMethod: false // Only retry in the `fixPaymentMethod` context.
+      )
     }
 
-    self.showWebHelp = self.pledgeDisclaimerViewDidTapLearnMoreSignal.mapConst(.trust)
+    self.paymentMethodsViewHidden = Signal.combineLatest(isLoggedIn, context)
+      .map { !$0 || $1.paymentMethodsViewHidden }
```

Comment:
Nit: Slightly hard to read with `$0`, would prefer that these variables were named so we had `!isLoggedIn || context.paymentMethodsViewHidden`

Reply:
Good point, I'll update it!